### PR TITLE
Storage [azdatalake] ETAG casing issue

### PR DESCRIPTION
**Pull request by tanyasethi-msft**
Fixes https://github.com/Azure/azure-rest-api-specs/issues/29233
Minor changes in multiple files are due to GoLintCLI warnings that were found in pipeline runs after this change https://github.com/Azure/azure-sdk-for-go/pull/23013

- [x] The purpose of this PR is explained in this or a referenced issue.
- [x] The PR does not update generated files.
   - These files are managed by the codegen framework at [Azure/autorest.go][].
- [x] Tests are included and/or updated for code changes.
- [x] Updates to module CHANGELOG.md are included.
- [x] MIT license headers are included in each file.

[Azure/autorest.go]: https://github.com/Azure/autorest.go

### CODE REVIEW

File: sdk/storage/azdatalake/internal/generated/zz_models_serde.go [hunk 1]
```diff
@@ -252,7 +252,7 @@ func (p *Path) UnmarshalJSON(data []byte) error {
 		case "creationTime":
 			err = unpopulate(val, "CreationTime", &p.CreationTime)
 			delete(rawMsg, key)
-		case "eTag":
+		case "etag":
```
Comment (jhendrixMSFT) [inline sdk/storage/azdatalake/internal/generated/zz_models_serde.go:255]: This changes the expectation in the JSON payload. Is it expected?

Comment (jhendrixMSFT) [reply sdk/storage/azdatalake/internal/generated/zz_models_serde.go:255]: Saw the linked issue.

File: sdk/storage/azdatalake/filesystem/client_test.go [hunk 2]
```diff
@@ -1471,6 +1471,43 @@ func (s *RecordedTestSuite) TestFilesystemListPathsWithRecursive() {
...
+	_require.NoError(err)
+
+	pager := fsClient.NewListPathsPager(true, nil)
+	for pager.More() {
+		resp, err := pager.NextPage(context.Background())
+		_require.NoError(err)
+
+		if err != nil {
+			break
+		}
```
Comment (souravgupta-msft) [inline sdk/storage/azdatalake/filesystem/client_test.go:1503]: Duplicate and can be removed. The above statement already checks the error.

Comment (tanyasethi-msft) [reply sdk/storage/azdatalake/filesystem/client_test.go]: Taken care

**Pull request merged**